test/libdnf/module/test_module.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "test_module.hpp"
+
+#include "utils.hpp"
+#include "utils/fs/file.hpp"
+
+#include "libdnf/module/module_errors.hpp"
+#include "libdnf/module/module_package.hpp"
+#include "libdnf/module/module_package_container.hpp"
+#include "libdnf/utils/format.hpp"
+
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+CPPUNIT_TEST_SUITE_REGISTRATION(ModuleTest);
+
+
+using namespace libdnf::module;
+
+
+void ModuleTest::test_load() {
+    std::ifstream module_yaml_file(PROJECT_SOURCE_DIR "/test/data/repos-repomd/repomd-modules/repodata/modules.yaml");
+    std::string module_yaml_string;
+    std::string line;
+    if (module_yaml_file.is_open()) {
+        while (getline(module_yaml_file, line)) {
+            module_yaml_string.append(line).append("\n");
+        }
+        module_yaml_file.close();
+    }
+
+    auto module_package_container = new ModulePackageContainer();
+    module_package_container->add(module_yaml_string);
+    CPPUNIT_ASSERT_EQUAL(11lu, module_package_container->modules.size());
```

Comment:
The test can ensure that all string representatives are in container, but it is a minor suggestion.

Reply:
I added assertions for one of the modules and for the others being present.